### PR TITLE
Regenerated logging service

### PR DESCRIPTION
**Pull request by michaelbausor**
Regenerated java logging service
Updated pom to reference gax 0.0.9

### CODE REVIEW

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java [hunk 1]
```diff
@@ -95,181 +94,68 @@
...
+        settingsBuilder.getClientLibVersion());
+
+    listLogMetricsSettings = settingsBuilder.listLogMetricsSettings().build();
+    getLogMetricSettings = settingsBuilder.getLogMetricSettings().build();
+    createLogMetricSettings = settingsBuilder.createLogMetricSettings().build();
+    updateLogMetricSettings = settingsBuilder.updateLogMetricSettings().build();
+    deleteLogMetricSettings = settingsBuilder.deleteLogMetricSettings().build();
   }
 
-  private static PageDescriptor<ListLogMetricsRequest, ListLogMetricsResponse, LogMetric>
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java]: Codacy Issue found: [The field name indicates a constant but its modifiers do not](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714612/issues/source?bid=3123104&fileBranchId=3228843#l156)

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java [hunk 2]
```diff
@@ -290,4 +176,170 @@ public Object extractNextToken(ListLogMetricsResponse payload) {
               return payload.getMetricsList();
             }
           };
+
+  public static class Builder extends ServiceApiSettings.Builder {
+    private final ImmutableList<ApiCallSettings.Builder> methodSettingsBuilders;
+
+    private PageStreamingCallSettings.Builder<
+            ListLogMetricsRequest, ListLogMetricsResponse, LogMetric>
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java]: Codacy Issue found: [Private field 'listLogMetricsSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714612/issues/source?bid=3123104&fileBranchId=3228843#l185)

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java [hunk 3]
```diff
@@ -290,4 +176,170 @@ public Object extractNextToken(ListLogMetricsResponse payload) {
...
+        listLogMetricsSettings;
+    private SimpleCallSettings.Builder<GetLogMetricRequest, LogMetric> getLogMetricSettings;
+    private SimpleCallSettings.Builder<CreateLogMetricRequest, LogMetric> createLogMetricSettings;
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java]: Codacy Issue found: [Private field 'updateLogMetricSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714612/issues/source?bid=3123104&fileBranchId=3228843#l188)

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java [hunk 4]
```diff
@@ -290,4 +176,170 @@ public Object extractNextToken(ListLogMetricsResponse payload) {
...
             }
           };
+
+  public static class Builder extends ServiceApiSettings.Builder {
+    private final ImmutableList<ApiCallSettings.Builder> methodSettingsBuilders;
+
+    private PageStreamingCallSettings.Builder<
+            ListLogMetricsRequest, ListLogMetricsResponse, LogMetric>
+        listLogMetricsSettings;
+    private SimpleCallSettings.Builder<GetLogMetricRequest, LogMetric> getLogMetricSettings;
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java]: Codacy Issue found: [Private field 'createLogMetricSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714612/issues/source?bid=3123104&fileBranchId=3228843#l187)

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java [hunk 5]
```diff
@@ -290,4 +176,170 @@ public Object extractNextToken(ListLogMetricsResponse payload) {
...
+    private SimpleCallSettings.Builder<CreateLogMetricRequest, LogMetric> createLogMetricSettings;
+    private SimpleCallSettings.Builder<UpdateLogMetricRequest, LogMetric> updateLogMetricSettings;
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java]: Codacy Issue found: [Private field 'deleteLogMetricSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714612/issues/source?bid=3123104&fileBranchId=3228843#l189)

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java [hunk 6]
```diff
@@ -290,4 +176,170 @@ public Object extractNextToken(ListLogMetricsResponse payload) {
               return payload.getMetricsList();
             }
           };
+
+  public static class Builder extends ServiceApiSettings.Builder {
+    private final ImmutableList<ApiCallSettings.Builder> methodSettingsBuilders;
+
+    private PageStreamingCallSettings.Builder<
+            ListLogMetricsRequest, ListLogMetricsResponse, LogMetric>
+        listLogMetricsSettings;
```
Comment (mziccard) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/MetricsServiceV2Settings.java]: Codacy Issue found: [Private field 'getLogMetricSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714612/issues/source?bid=3123104&fileBranchId=3228843#l186)

File: gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java [hunk 7]
```diff
@@ -369,4 +213,178 @@ public Object extractNextToken(ListMonitoredResourceDescriptorsResponse payload)
...
           };
+
+  public static class Builder extends ServiceApiSettings.Builder {
+    private final ImmutableList<ApiCallSettings.Builder> methodSettingsBuilders;
+
+    private SimpleCallSettings.Builder<DeleteLogRequest, Empty> deleteLogSettings;
+    private SimpleCallSettings.Builder<WriteLogEntriesRequest, WriteLogEntriesResponse>
+        writeLogEntriesSettings;
+    private PageStreamingCallSettings.Builder<
+            ListLogEntriesRequest, ListLogEntriesResponse, LogEntry>
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java]: Codacy Issue found: [Private field 'listLogEntriesSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714616/issues/source?bid=3123104&fileBranchId=3228843#l225)

File: gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java [hunk 8]
```diff
@@ -369,4 +213,178 @@ public Object extractNextToken(ListMonitoredResourceDescriptorsResponse payload)
...
+        listLogEntriesSettings;
+    private PageStreamingCallSettings.Builder<
+            ListMonitoredResourceDescriptorsRequest, ListMonitoredResourceDescriptorsResponse,
+            MonitoredResourceDescriptor>
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java]: Codacy Issue found: [Private field 'listMonitoredResourceDescriptorsSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714616/issues/source?bid=3123104&fileBranchId=3228843#l229)

File: gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java [hunk 9]
```diff
@@ -312,37 +182,11 @@ public Object extractNextToken(ListLogEntriesResponse payload) {
...
-              return payload.getResumeToken();
-            }
-
-            @Override
-            public Iterable<LogEntry> extractResources(ReadLogEntriesResponse payload) {
-              return payload.getEntriesList();
-            }
-          };
-
-  private static PageDescriptor<
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java]: Codacy Issue found: [The field name indicates a constant but its modifiers do not](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714616/issues/source?bid=3123104&fileBranchId=3228843#l185)

File: gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java [hunk 10]
```diff
@@ -369,4 +213,178 @@ public Object extractNextToken(ListMonitoredResourceDescriptorsResponse payload)
               return payload.getResourceDescriptorsList();
             }
           };
+
+  public static class Builder extends ServiceApiSettings.Builder {
+    private final ImmutableList<ApiCallSettings.Builder> methodSettingsBuilders;
+
```
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Settings.java]: Codacy Issue found: [Private field 'deleteLogSettings' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2388714616/issues/source?bid=3123104&fileBranchId=3228843#l220)

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java [hunk 11]
```diff
@@ -53,105 +53,13 @@
 
 // AUTO-GENERATED DOCUMENTATION AND SERVICE - see instructions at the top of the file for editing.
 /**
- * Service Description: See src/api/google/logging/v2/logging.proto for documentation
+ * Service Description:
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java]: "Service Description:" does not look like a nice class level javadoc.
We should ask the service owners to provide description in the service level (or having the generators meta-information include that).

Also, we should add a `@see` entry to https://github.com/googleapis/googleapis/blob/master/google/logging/v2/logging_config.proto

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java [hunk 12]
```diff
@@ -162,61 +70,120 @@ public static final String parseSinkFromSinkPath(String sinkPath) {
...
+  public static final ConfigServiceV2Api create(ConfigServiceV2Settings settings)
+      throws IOException {
     return new ConfigServiceV2Api(settings);
   }
 
   /**
-   * Constructs an instance of ConfigServiceV2Api, using the given settings. This is protected so
-   * that it easy to make a subclass, but otherwise, the static factory methods should be preferred.
+   * Constructs an instance of ConfigServiceV2Api, using the given settings.
+   * This is protected so that it easy to make a subclass, but otherwise, the static
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java]: I would replace "make" with "create"

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Settings.java [hunk 13]
```diff
@@ -55,18 +54,18 @@
...
 import io.grpc.Status;
+import java.io.IOException;
+import java.util.concurrent.ScheduledExecutorService;
+import org.joda.time.Duration;
 
 // Manually-added imports: add custom (non-generated) imports after this point.
 
 // AUTO-GENERATED DOCUMENTATION AND CLASS - see instructions at the top of the file for editing.
 @javax.annotation.Generated("by GAPIC")
 public class ConfigServiceV2Settings extends ServiceApiSettings {
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Settings.java]: generator should include class level javadoc.

Also, though redundant I would consider making it final (as it looks like you are not expecting it to be subclassed having a private constructor).

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Settings.java [hunk 14]
```diff
@@ -95,177 +94,68 @@
...
-  /**
-   * Returns the ApiCallableBuilder for the API method deleteSink.
-   *
-   * <!-- manual edit -->
-   * <!-- end manual edit -->
-   */
-  public ApiCallableBuilder<DeleteSinkRequest, Empty> deleteSinkMethod() {
-    return methods.deleteSinkMethod;
+  private ConfigServiceV2Settings(Builder settingsBuilder) throws IOException {
+    super(
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Settings.java]: I can't imagine the formatter that you are using did that?

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java [hunk 15]
```diff
@@ -311,24 +306,40 @@ public WriteLogEntriesResponse writeLogEntries(WriteLogEntriesRequest request) {
...
 
   // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
   /**
-   * Lists log entries.
-   * Use this method to examine log entries from Cloud Logging.
-   * See [Exporting Logs](/logging/docs/export) for other ways to copy
-   * log entries out of Cloud Logging.
+   * Lists log entries.  Use this method to retrieve log entries from Cloud
+   * Logging.  For ways to export log entries, see
+   * [Exporting Logs](/logging/docs/export).
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java]: We should cleanup or fix this "see" reference.

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java [hunk 16]
```diff
@@ -311,24 +306,40 @@ public WriteLogEntriesResponse writeLogEntries(WriteLogEntriesRequest request) {
    *
    * <!-- manual edit -->
    * <!-- end manual edit -->
+   * @throws ApiException if the remote call fails
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java]: Forgot to mention that in the earlier PR.

This method will **never** throw, right?

I think you should fix the ApiCallable javadoc to make it clear that its synchronous method, `call(...)` can throw `ApiException` and that the asynchronous ones, `futureCall` returns a future that can fail with `ApiException`.

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java [hunk 17]
```diff
@@ -311,24 +306,40 @@ public WriteLogEntriesResponse writeLogEntries(WriteLogEntriesRequest request) {
...
+   * @param filter Optional. An [advanced logs filter](/logging/docs/view/advanced_filters).
+   * The filter is compared against all log entries in the projects specified by
+   * `projectIds`.  Only entries that match the filter are retrieved.  An empty
+   * filter matches all log entries.
+   * @param orderBy Optional. How the results should be sorted.  Presently, the only permitted
+   * values are `"timestamp"` (default) and `"timestamp desc"`.  The first
+   * option returns entries in order of increasing values of
+   * `LogEntry.timestamp` (oldest first), and the second option returns entries
+   * in order of decreasing timestamps (newest first).  Entries with equal
+   * timestamps are returned in order of `LogEntry.insertId`.
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java]: replace `'LogEntry.insertId'` with `{@code LogEntry.insertId}`

File: gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java [hunk 18]
```diff
@@ -340,106 +351,49 @@ public WriteLogEntriesResponse writeLogEntries(WriteLogEntriesRequest request) {
 
   // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
   /**
-   * Lists log entries.
-   * Use this method to examine log entries from Cloud Logging.
-   * See [Exporting Logs](/logging/docs/export) for other ways to copy
-   * log entries out of Cloud Logging.
+   * Lists log entries.  Use this method to retrieve log entries from Cloud
+   * Logging.  For ways to export log entries, see
+   * [Exporting Logs](/logging/docs/export).
```
Comment (aozarov) [inline gcloud-java-logging/baseline/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java]: Same as above.

File: gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java [hunk 19]
```diff
@@ -227,8 +194,12 @@ public void close() throws IOException {
    *
    * <!-- manual edit -->
    * <!-- end manual edit -->
+   *
+   * @param projectName Required. The resource name of the project containing the sinks.
+   * Example: `"projects/my-logging-project"`, `"projects/01234567890"`.
```
Comment (aozarov) [inline gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java]: replace the "`" with "{@code"

**Comment by aozarov** [comment]
Looks good. I mentioned few more issues with the generated code (and gax) which I think you should fix in separate PRs (and create issues for). I want to say, again, that the flat baseline is a PITA.
Feel free to merge.

**Pull request merged**